uniffi_core/src/ffi/rustfuture/mod.rs

```diff
@@ -30,6 +30,32 @@ pub enum RustFuturePoll {
 /// to continue progress on the future.
 pub type RustFutureContinuationCallback = extern "C" fn(callback_data: u64, RustFuturePoll);
 
+/// This marker trait allows us to put different bounds on the `Future`s we support, based on
+/// `#[cfg()]` configuration.
+pub trait UniffiCompatibleFuture<T, E>: Future<Output = Result<T, E>> {}
+
+#[cfg(not(target_arch = "wasm32"))]
+impl<T, F, E> UniffiCompatibleFuture<T, E> for F where F: Future<Output = Result<T, E>> + Send {}
+
+/// # Safety:
+///
```

Comment:
Nit: How about mentioning the `Send` bound at the top?  The comment is long enough that readers might not see the actual type alias.  Maybe something like "Future's on WASM32 are not `Send` because it's a single threaded environment".